pages/createemotiontype.js

```diff
@@ -0,0 +1,33 @@
+import CreateEmotionTypeForm from "@/components/CreateEmotionTypeForm";
+import Heading from "@/components/Heading";
+import { useRouter } from "next/router";
+import { useState } from "react";
+
+export default function CreateEmotionType({
+  onCreateEmotionType,
+  emotionTypes,
+}) {
+  const [selectedEmotionType, setSelectedEmotionType] = useState("");
+  const router = useRouter();
+
+  const handleEmotionTypeSubmit = (formData) => {
+    setSelectedEmotionType(formData.emotionType);
+    router.push({
+      pathname: "/",
+      query: { showForm: "true", selectedEmotionType: formData.emotionType },
+    });
+  };
```

Comment:
since you redirect your user directly after setting the state, the state is lost ( the component is not visible anymore, you are on another page). therefore you don't need this state.